src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -0,0 +1,1461 @@
+     *
+     * @jls 15.4 Floating-point Expressions
+     */
+    public static Float16 subtract(Float16 minuend, Float16 subtrahend) {
+        return valueOf(minuend.floatValue() - subtrahend.floatValue());
+    }
+
+    /**
+     * Multiplies two {@code Float16} values as per the {@code *}
+     * operator semantics using the round to nearest rounding policy.
+     *
+     * The handling of signed zeros, NaNs, and infinities, other
+     * special cases by this method is the same as for the handling of
+     * those cases by the built-in {@code *} operator for
+     * floating-point multiplication (JLS {@jls 15.17.1}).
+     *
+     * @apiNote This method corresponds to the multiplication
+     * operation defined in IEEE 754.
+     *
+     * @param multiplier the first operand
+     * @param multiplicand the second operand
+     * @return the product of the operands
+     *
+     * @jls 15.4 Floating-point Expressions
+     */
+    public static Float16 multiply(Float16 multiplier, Float16 multiplicand) {
+        return valueOf(multiplier.floatValue() * multiplicand.floatValue());
+    }
+
+    /**
+     * Divides two {@code Float16} values as per the {@code /}
+     * operator semantics using the round to nearest rounding policy.
+     *
+     * The handling of signed zeros, NaNs, and infinities, other
+     * special cases by this method is the same as for the handling of
+     * those cases by the built-in {@code /} operator for
+     * floating-point division (JLS {@jls 15.17.2}).
+     *
+     * @apiNote This method corresponds to the division
+     * operation defined in IEEE 754.
+     *
+     * @param dividend the first operand
+     * @param divisor the second operand
+     * @return the quotient of the operands
+     *
+     * @jls 15.4 Floating-point Expressions
+     */
+    public static Float16 divide(Float16 dividend, Float16 divisor) {
+        return valueOf(dividend.floatValue() / divisor.floatValue());
+    }
+
+    /**
+     * {@return the square root of the operand} The square root is
+     * computed using the round to nearest rounding policy.
+     *
+     * The handling of zeros, NaN, infinities, and negative arguments
+     * by this method is analogous to the handling of those cases by
+     * {@link Math#sqrt(double)}.
+     *
+     * @apiNote
+     * This method corresponds to the squareRoot operation defined in
+     * IEEE 754.
+     *
+     * @param radicand the argument to have its square root taken
+     *
+     * @see Math#sqrt(double)
+     */
+    public static Float16 sqrt(Float16 radicand) {
+        // Rounding path of sqrt(Float16 -> double) -> Float16 is fine
+        // for preserving the correct final value. The conversion
+        // Float16 -> double preserves the exact numerical value. The
+        // of the double -> Float16 conversion also benefits from the
+        // 2p+2 property of IEEE 754 arithmetic.
+        return valueOf(Math.sqrt(radicand.doubleValue()));
```

Comment:
Kindly also add a reference to link you shared on lworld+fp16 branch.
https://dl.acm.org/doi/pdf/10.1145/221332.221334

Reply:
That reference is already cited around line 850.